### PR TITLE
Add sqlite3 support for building docker_cli

### PR DESCRIPTION
**Pull request by YetAnotherMinion**
I wanted to use `diesel_cli` with sqlite3 on an alpine linux box, so I added a build of `libsqlite3.a` to the docker image.

### CODE REVIEW

**Comment by clux** [comment]
Hey, thanks for this. Looks sensible, but need some tests here.

Maybe it works for diesel_cli itself, but I tried by adding a small test crate similar to the dieselcrate in the tests directory (basically just changing everything to sqlite over postgres) and that causes this to give relocation errors at the link stage.

```sh
  = note: /usr/bin/ld: /volume/target/debug/deps/liblibsqlite3_sys-13da0871e20cc8c2.rlib(sqlite3.o): relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
```

here's the source i had:

```rust
#[macro_use]
extern crate diesel;
#[macro_use]
extern crate diesel_codegen;

mod schema {
  table! {
      posts (id) {
          id -> Int4,
          title -> Varchar,
          body -> Text,
          published -> Bool,
      }
  }
}

mod models {
  use schema::posts;
  #[derive(Queryable)]
  pub struct Post {
      pub id: i32,
      pub title: String,
      pub body: String,
      pub published: bool,
  }

  // apparently this can be done without heap storage, but lifetimes spread far..
  #[derive(Insertable)]
  #[table_name="posts"]
  pub struct NewPost {
      pub title: String,
      pub body: String,
  }
}

use diesel::prelude::*;
use diesel::sqlite::SqliteConnection;

fn main() {
    let database_url = std::env::var("DATABASE_URL").unwrap_or("main.db".into());
    SqliteConnection::establish(&database_url).unwrap();
}
```

with toml:

```toml
[package]
authors = ["clux <sszynrae@gmail.com>"]
name = "dieselcratelite"
version = "0.1.0"

[dependencies.diesel]
features = ["sqlite"]
version = "0.16.0"

[dependencies.diesel_codegen]
features = ["sqlite"]
version = "0.16.0"
```

**Comment by YetAnotherMinion** [comment]
Test crate now passes. I added `-fPIC -pie` to the compiler arguments and rebuilt the container. Added a test crate for dieselsqlite as well.

Squashed everything into the same commit so you'll have to force pull the branch.

**Comment by clux** [comment]
Ah, that sounds sensible.

File: test/dieselpgcrate/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 authors = ["clux <sszynrae@gmail.com>"]
-name = "dieselcrate"
+name = "dieselpgcrate"
```
Comment (clux) [inline test/dieselpgcrate/Cargo.toml]: this rename probably breaks the makefile `make test` should work, worth adding a target for the new test crate as well!

File: test/dieselsqlitecrate/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+[dependencies.diesel]
+features = ["sqlite"]
+version = "0.16.0"
+
+[dependencies.diesel_codegen]
+features = ["sqlite"]
+version = "0.16.0"
+
+[patch.crates-io]
+pq-sys = { git = 'https://github.com/golddranks/pq-sys' }
```
Comment (clux) [inline test/dieselsqlitecrate/Cargo.toml]: you don't need pq-sys in this toml file

**Comment by clux** [comment]
I've fixed CI now, so you should have a build if you rebase with master.

**Comment by YetAnotherMinion** [comment]
I completely missed the Makefile, was manually invoking the test.sh script for every crate like a fool... ;).

The hyper crate fails `make test-hyper`
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Error { repr: Custom(Custom { kind: Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) })', /checkout/src/libcore/result.rs:916:4
```

But it also fails on your previous commit 44fb394b343d192449d72cfb0679982a8855a153, so I did not introduce the error. I think it is because the binary is run without setting SSL_CERT_* env variables.

**Comment by clux** [comment]
Huh, that's odd. I tried a clean new build on master now after cleaning out everything and it worked for me. The SSL_CERT variable is set in main.rs so that's how it's been working before at least. I'll try restarting your build.

File: Makefile [hunk 3]
```diff
@@ -33,8 +35,9 @@ clean-lock:
 	sudo find . -iname Cargo.lock -exec rm {} \;
 clean-builds:
 	sudo find . -mindepth 3 -maxdepth 3 -name target -exec rm -rf {} \;
+	rm test/dieselsqlitecrate/main.db
```
Comment (clux) [inline Makefile]: ah, thanks

File: Makefile [hunk 4]
```diff
@@ -18,8 +18,10 @@ test-rocket:
 	./test.sh rocket
 test-pq:
 	./test.sh pq
-test-diesel:
-	./test.sh diesel
+test-dieselpg:
+	./test.sh dieselpg
+test-dieselsqlite:
+	./test.sh dieselsqlite
```
Comment (clux) [inline Makefile]: :+1:

File: Dockerfile [hunk 5]
```diff
@@ -100,6 +101,15 @@ RUN curl -sSL https://ftp.postgresql.org/pub/source/v$PQ_VER/postgresql-$PQ_VER.
     rm $PREFIX/lib/*.so && rm $PREFIX/lib/*.so.* && rm $PREFIX/lib/postgres* -rf &&  \
     cd .. && rm -rf postgresql-$PQ_VER
 
+# Build libsqlite3 using same configuration as the alpine linux main/sqlite package
```
Comment (clux) [inline Dockerfile]: straight out of the https://git.alpinelinux.org/cgit/aports/tree/main/sqlite/APKBUILD?h=3.7-stable i presume?

**Comment by clux** [comment]
yeah, somehow rocket on your branch is sad, not sure why, it works here.

File: Dockerfile (hunk 5 above)
Comment (YetAnotherMinion) [reply Dockerfile]: Yes.

Comment (YetAnotherMinion) [reply Dockerfile]: However I did not include the rpath removal because we are only using the static library portion of the build.

**Comment by clux** [comment]
Right. I've bumped those test dependencies, master passes again. Yours probably will if you rebase :-)

**Comment by clux** [comment]
Awesome, seems to have passed now. Thanks a lot for this :-) I'll add some notes on the new compatibility in the readme.

Hopefully this will just keep building without any issues in the future.

**Pull request merged**